### PR TITLE
refactor config; remove random simulation variables and override_dict

### PR DESCRIPTION
**Pull request by dpaiton**

### CODE REVIEW

File: src/elfpy/simulators.py [hunk 1]
```diff
@@ -54,20 +49,11 @@ def __init__(
         self.block_number = 0
         self.daily_block_number = 0
         seconds_in_a_day = 86400
-        self.time_between_blocks = seconds_in_a_day / self.config.simulator.num_blocks_per_day
+        self.time_between_blocks = seconds_in_a_day / self.config.num_blocks_per_day
         self.run_trade_number = 0
         self.start_time: datetime.datetime | None = None
         self.simulation_state = SimulationState()
 
-    def check_vault_apr(self) -> None:
```
Comment (dpaiton) [inline src/elfpy/simulators.py:62]: moved to config type

File: src/elfpy/simulators.py [hunk 2]
```diff
@@ -81,12 +67,6 @@ def set_rng(self, rng: Generator) -> None:
             raise TypeError(f"rng type must be a random number generator, not {type(rng)}.")
         self.rng = rng
 
-    def log_config_variables(self) -> None:
```
Comment (dpaiton) [inline src/elfpy/simulators.py:84]: moved to config __str__

File: tests/test_market.py [hunk 3]
```diff
@@ -239,7 +235,7 @@ def test_100_open_short(self):
         # assign to appropriate token, for readability using absolute values, assigning +/- below
         d_base = trade_result  # proceeds from your sale of bonds, go into your margin account so you don't rug
         d_bonds = 100
-        open_share_price = 1.0086194128439765
+        open_share_price = 1.0
```
Comment (dpaiton) [inline tests/test_market.py:238]: Don't know why I had to change this :-/

File: tests/test_sim.py [hunk 4]
```diff
@@ -7,142 +7,69 @@
...
+    """Test running a simulation using each pricing model type"""
 
     @staticmethod
     def setup_logging(log_level=logging.DEBUG):
         """Setup logging and handlers for the test"""
         log_filename = ".logging/test_sim.log"
         output_utils.setup_logging(log_filename, log_level=log_level)
 
-    @staticmethod
-    def setup_config(config_file, override_dict=None) -> Config:
```
Comment (dpaiton) [inline tests/test_sim.py:30]: don't need this test anymore

File: tests/test_sim.py [hunk 5]
```diff
@@ -7,142 +7,69 @@
...
-from elfpy.simulators import Simulator
-from elfpy.utils.config import Config
-from elfpy.utils.parse_config import load_and_parse_config_file
+from elfpy.types import Config
 from elfpy.utils import sim_utils  # utilities for setting up a simulation
 import elfpy.utils.outputs as output_utils
-import elfpy.utils.parse_config as config_utils
 
 
-class BaseSimTest(unittest.TestCase):
```
Comment (dpaiton) [inline tests/test_sim.py:18]: moved this down to the TestSimulator def

**Review by sentilesdal — APPROVED** [review]

**Pull request merged**